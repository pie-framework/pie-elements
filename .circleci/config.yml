version: 2

defaults: &defaults
  working_directory: ~/repo
  docker:
    - image: circleci/node:11.10.0

reg_auth: &reg_auth
  name: Authenticate with registry
  command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/repo/.npmrc

restore_cache: &restore_cache
  keys:
    - v2-dependencies-{{ checksum "package.json" }}
    # fallback to using the latest cache if no exact match is found
    - v2-dependencies-

save_cache: &save_cache
  paths:
    - node_modules
  key: v2-dependencies-{{ checksum "package.json" }}

jobs:
  lint:
    <<: *defaults
    steps:
      - checkout
      - run: *reg_auth
      - restore_cache: *restore_cache
      - run: yarn install
      - save_cache: *save_cache
      - run: scripts/build lint
  test:
    <<: *defaults
    steps:
      - checkout
      - run: *reg_auth
      - restore_cache: *restore_cache
      - run: yarn install
      - save_cache: *save_cache
      - run: scripts/build babel test
  build:
    <<: *defaults
    steps:
      - checkout
      - run: *reg_auth
      - restore_cache: *restore_cache
      - run: yarn install
      - save_cache: *save_cache
      - run: scripts/build release --next --token $NPM_TOKEN
  release:
    <<: *defaults
    steps:
      - checkout
      - run: *reg_auth
      - restore_cache: *restore_cache
      - run: yarn install
      - save_cache: *save_cache
      - run: scripts/build release --token $NPM_TOKEN

workflows:
  version: 2

  simple_build:
    jobs:
      - lint:
          filters:
            branches:
              ignore:
                - develop
                - master
      - test:
          filters:
            branches:
              ignore:
                - develop
                - master
  build:
    jobs:
      - build:
          context: jira
          filters:
            branches:
              only:
                - develop

  release:
    jobs:
      - release:
          context: jira
          filters:
            branches:
              only:
                - master
  # nightly:
  #   triggers:
  #     - schedule:
  #         cron: "0 0 * * *"
  #         filters:
  #           branches:
  #             only:
  #               - develop
  #   jobs:
  #     - build
