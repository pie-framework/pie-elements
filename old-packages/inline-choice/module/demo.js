import Configure from './configure.js';
import Element from './element.js';
import * as controller from './controller.js';

var model = (id, element) => ({
  id,
  element,

  defaultLang: 'en-US',
  choiceLabel: 'Select option ...',
  choices: [
    {
      correct: true,
      value: 'sweden',
      label: 'Sweden',
    },
    {
      value: 'iceland',
      label: 'Iceland',
      feedback: {
        type: 'default',
      },
    },
    {
      value: 'norway',
      label: 'Norway',
    },
    {
      value: 'finland',
      label: 'Finland',
      feedback: {
        type: 'custom',
        value: 'Nokia was founded in Finland.',
      },
    },
  ],
});

var generate = {
  model: model,
};

const { model: model$1 } = generate;

var config = {
  elements: {
    'inline-choice': '../..',
  },
  models: [model$1('1', 'inline-choice')],
};

//Note: demo-el is a custom element loaded in the markup.
customElements.whenDefined('demo-el').then(() => {
  config.models.forEach((m) => {
    const de = document.createElement('demo-el');
    document.body.appendChild(de);
    de.def = { tagName: m.element, Element, Configure, controller };
    de.model = m;
  });
});
