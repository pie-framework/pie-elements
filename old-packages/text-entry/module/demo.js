import Configure from './configure.js';
import Element from './element.js';
import * as controller from './controller.js';

var model = (id, element) => ({
  id,
  element,
  feedback: {
    correct: {
      type: 'custom',
      custom: 'correct-o',
    },
    incorrect: {
      type: 'custom',
      custom: 'custom feedback',
    },
    partial: {
      type: 'custom',
      custom: 'foo',
    },
  },
  correctResponses: {
    values: ['mutt', 'hound'],
    ignoreWhitespace: true,
    ignoreCase: false,
  },
  partialResponses: {
    values: ['mutty'],
    ignoreWhitespace: true,
    ignoreCase: true,
    awardPercentage: '50',
  },
  answerBlankSize: '10',
  answerAlignment: 'left',
  prompt: 'Question Prompt goes here',
  allowDecimal: true,
  allowIntegersOnly: false,
  allowThousandsSeparator: true,
});

var generate = {
  model: model,
};

const { model: model$1 } = generate;

var config = {
  elements: {
    'text-entry': '../..',
  },
  models: [model$1('1', 'text-entry')],
};

//Note: demo-el is a custom element loaded in the markup.
customElements.whenDefined('demo-el').then(() => {
  config.models.forEach((m) => {
    const de = document.createElement('demo-el');
    document.body.appendChild(de);
    de.def = { tagName: m.element, Element, Configure, controller };
    de.model = m;
  });
});
