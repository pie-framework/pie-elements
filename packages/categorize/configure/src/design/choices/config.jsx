import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';

import TextField from '@material-ui/core/TextField';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import { Checkbox } from '@pie-lib/config-ui';

const positionOptions = [
  {
    label: 'Above',
    value: 'above',
    secondaryText: 'Choices will be shown above categories'
  },
  {
    label: 'Below',
    value: 'below',
    secondaryText: 'Choices will be shown below categories'
  },
  {
    label: 'Left',
    value: 'left',
    secondaryText: 'Choices will be shown to the left of the categories'
  },
  {
    label: 'Right',
    value: 'right',
    secondaryText: 'Choices will be shown to the right of the categories'
  }
];

export class Config extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    config: PropTypes.object,
    categoryCountIsOne: PropTypes.bool,
    allChoicesHaveCount: PropTypes.func,
    onModelChanged: PropTypes.func
  };

  static defaultProps = {};

  state = {
    anchorEl: null
  };

  toggleRemoveAllTiles = () => {
    const { config, onModelChanged } = this.props;

    const allAtOne = this.props.allChoicesHaveCount(1);
    const update = config.choices.map(c => {
      c.categoryCount = allAtOne ? 0 : 1;
      return c;
    });

    onModelChanged({ choices: update });
  };

  changeLabel = ({ target }) => {
    this.props.onModelChanged({ choicesLabel: target.value });
  };

  toggleShuffle = () => {
    this.props.onModelChanged({
      lockChoiceOrder: !this.props.config.lockChoiceOrder
    });
  };

  changePosition = position => {
    this.props.onModelChanged({ choicesPosition: position.value });

    this.setState({ anchorEl: null });
  };

  handleClickPosition = event => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleMenuItemClick = (event, index) => {
    this.setState({ selectedIndex: index, anchorEl: null });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  render() {
    const { anchorEl } = this.state;
    const { classes, className, categoryCountIsOne, config } = this.props;

    const positionOption =
      positionOptions.find(option => option.value === config.choicesPosition) ||
      positionOption[1];

    return (
      <div className={classNames(classes.config, className)}>
        <div className={classes.row}>
          <Checkbox
            label={'Remove all tiles after placing'}
            checked={categoryCountIsOne}
            onChange={this.toggleRemoveAllTiles}
          />
          <Checkbox
            label={'Lock Choice Order'}
            checked={config.lockChoiceOrder}
            onChange={this.toggleShuffle}
          />
        </div>

        <div className={classes.configuration}>
          <TextField
            className={classes.label}
            InputLabelProps={{
              shrink: true
            }}
            label="Label"
            value={config.choicesLabel}
            onChange={this.changeLabel}
          />
          <List component="nav">
            <ListItem
              button
              aria-haspopup="true"
              onClick={this.handleClickPosition}
            >
              <ListItemText
                primary="Choices Position"
                secondary={positionOption.secondaryText}
              />
            </ListItem>
          </List>
          <Menu
            id="lock-menu"
            anchorEl={anchorEl}
            open={Boolean(anchorEl)}
            onClose={this.handleClose}
          >
            {positionOptions.map(option => (
              <MenuItem
                key={option.value}
                selected={option.value === config.choicesPosition}
                onClick={() => this.changePosition(option)}
              >
                {option.label}
              </MenuItem>
            ))}
          </Menu>
        </div>
      </div>
    );
  }
}
const styles = theme => ({
  configuration: {
    display: 'grid',
    alignItems: 'center',
    gridColumnGap: `${theme.spacing.unit}px`,
    gridRowGap: `${theme.spacing.unit}px`,
    gridTemplateColumns: 'repeat(2, 1fr)',
    justifyContent: 'space-between',
    marginBottom: theme.spacing.unit
  },
  choiceConfig: {
    width: '100%',
    maxWidth: 360,
    backgroundColor: theme.palette.background.paper
  },
  row: {
    display: 'grid',
    gridColumnGap: `${theme.spacing.unit}px`,
    gridRowGap: `${theme.spacing.unit}px`,
    gridTemplateColumns: 'repeat(2, 1fr)',
    justifyContent: 'space-between',
    marginTop: theme.spacing.unit,
    marginBottom: theme.spacing.unit
  },
  shuffleCheckbox: {
    gridColumnStart: 2,
    gridColumnEnd: 2
  }
});
export default withStyles(styles)(Config);
