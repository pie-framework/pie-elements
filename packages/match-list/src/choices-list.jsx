import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import isEmpty from 'lodash/isEmpty';
import isUndefined from 'lodash/isUndefined';
import find from 'lodash/find';
import { DragAnswer } from './answer';
import { MatchDroppablePlaceholder } from '@pie-lib/drag';

export class ChoicesList extends React.Component {
  static propTypes = {
    classes: PropTypes.object,
    session: PropTypes.object.isRequired,
    instanceId: PropTypes.string.isRequired,
    model: PropTypes.object.isRequired,
    disabled: PropTypes.bool.isRequired,
  };

  render() {
    const { model, classes, disabled, session, instanceId, onRemoveAnswer } = this.props;
    const { config } = model;
    const { duplicates } = config;

    return (
      <>
        {MatchDroppablePlaceholder ? (
          <MatchDroppablePlaceholder disabled={disabled} onRemoveAnswer={onRemoveAnswer}>
            {config.answers
              .filter(
                (answer) =>
                  duplicates ||
                  isEmpty(session) ||
                  !session.value ||
                  isUndefined(find(session.value, (val) => val === answer.id))
              )
              .map((answer) => (
                <DragAnswer
                  key={answer.id}
                  instanceId={instanceId}
                  className={classes.choice}
                  draggable={true}
                  disabled={disabled}
                  session={session}
                  type={'choice'}
                  {...answer}
                />
              ))}
          </MatchDroppablePlaceholder>
        ) : (
          <div className={classes.answersContainer}>
            {' '}
            {config.answers
              .filter(
                (answer) =>
                  duplicates ||
                  isEmpty(session) ||
                  !session.value ||
                  isUndefined(find(session.value, (val) => val === answer.id))
              )
              .map((answer) => (
                <DragAnswer
                  key={answer.id}
                  instanceId={instanceId}
                  className={classes.choice}
                  draggable={true}
                  disabled={disabled}
                  session={session}
                  type={'choice'}
                  {...answer}
                />
              ))}{' '}
          </div>
        )}
      </>
    );
  }
}

const styles = () => ({
  answersContainer: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
    marginTop: 50
  },
  choice: {
    minHeight: '40px',
    minWidth: '200px',
    height: 'initial',
  },
});

export default withStyles(styles)(ChoicesList);
