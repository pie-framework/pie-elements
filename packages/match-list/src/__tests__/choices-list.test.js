import * as React from 'react';
import { shallow } from 'enzyme';
import { ChoicesList } from '../choices-list';
import { model } from '../../docs/demo/config';

describe('ChoicesList', () => {
  const defaultProps = {
    model: model('1'),
    session: {},
    classes: {},
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ChoicesList {...defaultProps} />);
  });

  describe('render', () => {
    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
