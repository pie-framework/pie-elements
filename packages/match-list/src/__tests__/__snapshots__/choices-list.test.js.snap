// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChoicesList render renders correctly 1`] = `
<Fragment>
  <div>
     
    <DragSource(WithStyles(Answer))
      draggable={true}
      id={1}
      key="1"
      session={Object {}}
      title="Answer 1"
      type="choice"
    />
    <DragSource(WithStyles(Answer))
      draggable={true}
      id={2}
      key="2"
      session={Object {}}
      title="Answer 2"
      type="choice"
    />
    <DragSource(WithStyles(Answer))
      draggable={true}
      id={3}
      key="3"
      session={Object {}}
      title="Answer 3"
      type="choice"
    />
    <DragSource(WithStyles(Answer))
      draggable={true}
      id={4}
      key="4"
      session={Object {}}
      title="Answer 4"
      type="choice"
    />
    <DragSource(WithStyles(Answer))
      draggable={true}
      id={5}
      key="5"
      session={Object {}}
      title="Answer 5"
      type="choice"
    />
    <DragSource(WithStyles(Answer))
      draggable={true}
      id={6}
      key="6"
      session={Object {}}
      title="Answer 6"
      type="choice"
    />
     
  </div>
</Fragment>
`;
